core/src/price_estimation/kraken.rs

```diff
@@ -91,19 +102,22 @@ where
             let asset_pairs: Vec<_> = token_asset_pairs.keys().map(String::as_str).collect();
             let ticker_infos = self.api.ticker(&asset_pairs).await?;
 
-            let prices = ticker_infos
-                .iter()
-                .flat_map(|(pair, info)| {
-                    let token = token_asset_pairs.get(pair)?;
-                    let price = self
-                        .token_info_fetcher
-                        .get_token_info(*token)
-                        .ok()?
-                        .get_owl_price(info.p.last_24h());
-
-                    Some((*token, price))
+            let prices = stream::iter(ticker_infos)
+                .filter_map(|(pair, info)| {
+                    let token_info_fetcher = self.token_info_fetcher.clone();
+                    let token = token_asset_pairs.get(&pair);
+                    async move {
+                        let price = token_info_fetcher
+                            .get_token_info(*(token?))
```

Comment:
It's weird that we do `get_token_info` on the same tokens that we already have fetched previously through `get_token_asset_pairs`. Doesn't really matter if it's cached though.

Reply:
Good point. I was able to get rid of it by putting the info into the asset pair hash map we collect already.